lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java

```diff
@@ -72,17 +75,30 @@ public ClassificationServiceImpl(TaskanaEngine taskanaEngine, ClassificationMapp
     }
 
     @Override
-    public void addClassification(Classification classification) {
-        LOGGER.debug("entry to addClassification(classification = {})", classification);
-       try {
+    public void createClassification(Classification classification) throws ClassificationAlreadyExistException {
+        LOGGER.debug("entry to createClassification(classification = {})", classification);
+        try {
             taskanaEngineImpl.openConnection();
-            classification.setId(IdGenerator.generateWithPrefix(ID_PREFIX_CLASSIFICATION));
-            classification.setCreated(Date.valueOf(LocalDate.now()));
-
-            this.setDefaultValues(classification);
-
-            classificationMapper.insert(classification);
-            LOGGER.debug("Method addClassification added classification {}.", classification);
+            ClassificationImpl classificationImpl = (ClassificationImpl) classification;
```

Comment:
we obtain a Classification from the client. Before we continue, we should ensure that it is a valid object. This implies for instance that the Id is set ...

Reply:
setDefaultValues() is called before the insert-call. setting an ID if is/was null is included there.